optaplanner-core/src/main/java/org/optaplanner/core/config/score/director/ScoreDirectorFactoryConfig.java

```diff
@@ -406,38 +441,40 @@ public ScoreDefinition buildDeprecatedScoreDefinition() {
             SolverConfigContext configContext, EnvironmentMode environmentMode,
             SolutionDescriptor<Solution_> solutionDescriptor) {
         AbstractScoreDirectorFactory<Solution_> easyScoreDirectorFactory = buildEasyScoreDirectorFactory(solutionDescriptor);
+        AbstractScoreDirectorFactory<Solution_> constraintStreamScoreDirectorFactory = buildConstraintStreamScoreDirectorFactory(solutionDescriptor);
         AbstractScoreDirectorFactory<Solution_> incrementalScoreDirectorFactory = buildIncrementalScoreDirectorFactory(solutionDescriptor);
         AbstractScoreDirectorFactory<Solution_> droolsScoreDirectorFactory = buildDroolsScoreDirectorFactory(configContext, solutionDescriptor);
-        AbstractScoreDirectorFactory<Solution_> scoreDirectorFactory;
-        if (easyScoreDirectorFactory != null) {
+        if (Stream.of(easyScoreDirectorFactory, constraintStreamScoreDirectorFactory,
+                incrementalScoreDirectorFactory, droolsScoreDirectorFactory)
+                .filter(Objects::nonNull).count() > 1) {
+            List<String> scoreDirectorFactoryPropertyList = new ArrayList<>(4);
```

Comment:
Yes, I just tried it and the code became less readable.
Because 4 will always be greater or equal to the count - and it's a temp list - I am leaving it as is.